plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestingThriftHttpMetastoreServer.java

```diff
@@ -0,0 +1,180 @@
+import com.google.inject.TypeLiteral;
+import io.airlift.bootstrap.Bootstrap;
+import io.airlift.bootstrap.LifeCycleManager;
+import io.airlift.http.server.HttpServerInfo;
+import io.airlift.http.server.TheServlet;
+import io.airlift.http.server.testing.TestingHttpServerModule;
+import io.airlift.node.testing.TestingNodeModule;
+import io.trino.hive.thrift.metastore.Database;
+import io.trino.hive.thrift.metastore.NoSuchObjectException;
+import io.trino.hive.thrift.metastore.ThriftHiveMetastore;
+import jakarta.servlet.Servlet;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.apache.thrift.TException;
+import org.apache.thrift.TProcessor;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.protocol.TProtocol;
+import org.apache.thrift.protocol.TProtocolFactory;
+import org.apache.thrift.transport.TIOStreamTransport;
+import org.apache.thrift.transport.TTransport;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static com.google.common.reflect.Reflection.newProxy;
+
+public class TestingThriftHttpMetastoreServer
+        implements Closeable
+{
+    private final TestingThriftHttpServlet thriftHttpServlet;
+    private final LifeCycleManager lifeCycleManager;
+    private final URI baseUri;
+
+    public TestingThriftHttpMetastoreServer(InMemoryThriftMetastore delegate, Consumer<HttpServletRequest> requestInterceptor)
+    {
+        ThriftHiveMetastore.Iface mockThriftHandler = proxyHandler(delegate, ThriftHiveMetastore.Iface.class);
+        TProcessor processor = new ThriftHiveMetastore.Processor<>(mockThriftHandler);
+        thriftHttpServlet = new TestingThriftHttpServlet(processor, new TBinaryProtocol.Factory(), requestInterceptor);
+        Bootstrap app = new Bootstrap(
+                new TestingNodeModule(),
+                new TestingHttpServerModule(),
+                binder -> {
+                    binder.bind(new TypeLiteral<Map<String, String>>() {}).annotatedWith(TheServlet.class).toInstance(ImmutableMap.of());
+                    binder.bind(Servlet.class).annotatedWith(TheServlet.class).toInstance(thriftHttpServlet);
+                });
+
+        Injector injector = app
+                .doNotInitializeLogging()
+                .initialize();
+
+        lifeCycleManager = injector.getInstance(LifeCycleManager.class);
+        HttpServerInfo httpServerInfo = injector.getInstance(HttpServerInfo.class);
+        baseUri = httpServerInfo.getHttpUri();
+    }
+
+    public static <T> T proxyHandler(InMemoryThriftMetastore delegate, Class<T> iface)
+    {
+        return newProxy(iface, (proxy, method, args) -> {
+            switch (method.getName()) {
```

Comment:
nit: We can replace with switch expression.

Reply:
The following code should work:
```java
    public static <T> T proxyHandler(InMemoryThriftMetastore delegate, Class<T> iface)
    {
        return newProxy(iface, (proxy, method, args) -> switch (method.getName()) {
            case "getAllDatabases" -> delegate.getAllDatabases();
            case "getDatabase" -> {
                Optional<Database> optionalDatabase = delegate.getDatabase(args[0].toString());
                yield optionalDatabase.orElseThrow(() -> new NoSuchObjectException(""));
            }
            default -> throw new UnsupportedOperationException();
        });
    }
```

Reply:
Thanks. Used your code snippet.